connectors/sources/onedrive.py

```diff
@@ -572,9 +709,14 @@ async def get_docs(self, filtering=None):
 
                 async for entity, download_url in self.client.get_owned_files(user_id):
                     entity = self.prepare_doc(entity)
+                    entity_permissions = await self.get_entity_permission(
+                        user_id=user_id, file_id=entity.get("_id")
+                    )
```

Comment:
Although `get_entity_permission`  is guarded by `if not self._dls_enabled():`, do you think it's better to wrap it in `_decorate_with_access_control`? 

There shouldn't be duplicated calls for a single entity unless the logic itself generating duplicated entity, for which the current code will result in duplicated calls to `get_entity_permission` as well.

Reply:
Yes! Better to wrap in `_decorate_with_access_control`